Tests/NIOTests/EventLoopTest.swift

```diff
@@ -173,6 +173,36 @@ public class EventLoopTest : XCTestCase {
         XCTAssertNil(weakEventLoop)
     }
 
+    public func testEventLoopGroupMakeIterator() throws {
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: System.coreCount)
+        defer {
+            XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully())
+        }
+
+        var counter = 0
+        eventLoopGroup.makeIterator()?.forEach { _ in
+            counter += 1
+        }
+
+        XCTAssertEqual(counter, System.coreCount)
+    }
+
+    public func testEventLoopMakeIterator() throws {
+        let eventLoop = EmbeddedEventLoop()
+        var iterator = eventLoop.makeIterator()
+        defer {
+            XCTAssertNoThrow(try eventLoop.syncShutdownGracefully())
+        }
+
+        var counter = 0
+        iterator?.forEach { loop in
+            XCTAssertTrue(loop === eventLoop)
+            counter += 1
+        }
+
+        XCTAssertEqual(counter, 1)
+    }
```

Comment:
Can you write two more tests for me? One that each `EventLoop` in the `MultiThreadedEventLoopGroup` itself is an iterator of one, and then one more where you write a dummy conformance to `EventLoopGroup` and prove that we get `nil`?

Reply:
Done, I added the first test to an existing one and created a new one for the nil check